### PR TITLE
Chi 1434 add media endpoint

### PR DESCRIPTION
**Pull request by stephenhand**
# Review after #466 

## Description

Add new endpoint to HRM to add conversationMedia to a contact

It's not very DRY, but since the same functionality will be removed from the create contact endpoint once old Flex versions are out of circulation, there didn't seem much point in factoring out the shared logic for a few weeks

No tests yet, wanted to make sure it was what we needed before investing that time, you can either review & get it merged without tests and the tests will be another PR, we can wait until the tests are in

### Checklist
- [X] Corresponding issue has been opened
- [ ] New tests added

### Related Issues
CHI-1434

### Verification steps

Automated tests (coming)

### CODE REVIEW

File: hrm-domain/hrm-service/src/contact/contact-routes-v0.ts [hunk 1]
```diff
@@ -146,4 +147,31 @@ contactsRouter.patch(
   },
 );
 
+contactsRouter.post(
+  '/:contactId/conversationMedia',
+  validatePatchPayload,
+  canEditContact,
```
Comment (GPaoloni) [inline hrm-domain/hrm-service/src/contact/contact-routes-v0.ts:153]: Good for now I think? :thinking:

File: hrm-domain/hrm-service/src/referral/referral-data-access.ts [hunk 2]
```diff
@@ -83,3 +84,12 @@ export const createReferralRecord =
       throw dbErr;
     }
   };
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const deleteContactReferrals =
```
Comment (GPaoloni) [inline hrm-domain/hrm-service/src/referral/referral-data-access.ts:89]: Used in a next PR?

**Review by GPaoloni — APPROVED** [review]
LGTM

File: hrm-domain/hrm-service/src/contact/contact-routes-v0.ts (hunk 1 above)
Comment (GPaoloni) [reply hrm-domain/hrm-service/src/contact/contact-routes-v0.ts:153]: I was thinking about this.. Right now the behavior is packed with the fact that a worker "can create a contact". Could exist the day where "create" and "edit" contacts have different rules, resulting in this endpoint failing when the counselor tries to "finalize the contact"? Could that lead to missing conversation medias? Just thinking out loud here.

Comment (stephenhand) [reply hrm-domain/hrm-service/src/contact/contact-routes-v0.ts:153]: You are correct, those additional handlers were added in error. This endpoint should be available to anyone who can add a contact

In a follow on PR when we add the 'finalizedAt' property, it should be forbidden on finalized contacts.

In the future I can see a scenario where conversationMedia isn't added from the front end at all, but off the back of task router events etc., then this endpoint could be not callable by flex users at all, but that's for the future

**Pull request merged**